0000-product.md

```diff
@@ -0,0 +1,463 @@
+Product Property Schema.
+
+``` 
+message Product { 
+    enum ProductNamespace { 
+        UNSET_NAMESPACE = 0; 
+        GS1 = 1; 
+    }
+    string product_id = 1; 
+    ProductNamespace product_namespace = 2; 
+    string owner = 3;
+    repeated PropertyValue properties = 4; 
+} 
+```
+
+The GS1 GTIN is an identifier (product_id) used to identify trade items.  A GTIN
+is the data transmitted from a barcode scan and is made up of a company prefix
+(or GS1-8 prefix) and item reference.  We will initially support GTIN-12,
+GTIN-13, and GTIN-14. (GTIN-8 may be supported in the future.) The GS1 GTIN
+specification is documented in section 3.3.2, Identification of a trade item
+(GTIN): AI (01), on page 140 of the **GS1 General Specification**:
+
+https://www.gs1.org/sites/default/files/docs/barcodes/GS1_General_Specifications.pdf
+
+The GS1-8 prefix is a unique string of 3 digits, and the GS1 company prefix
+consists of 4-12 digits.  The GS1-8 prefixes is issued by the GS1 Global Office.
+The GS1-8 prefix is allocated to a GS1 member organization to issue the GTIN-8’s
+to other areas.  The GS1 company prefix can be issued by the GS1 Global Office
+or a member organization. Usually issued to a company itself.
+
+Also relevant is the GS1-8 Prefix or GS1 Company Prefix which namespaces the
+GTIN by issuing organization.  GS1 manages the master data for company prefixes.
+Owning companies issue their own GTINs within their prefixed namespace.  Details
+about GS1 Company Prefixes can be found in section 1.4.4, GS1 Company Prefix, on
+page 20 of the GS1 General Specifications.
+
+### Referencing Products
+
+Products are uniquely referenced by their product_id and product_namespace.  For
+GS1, Products are referenced by the GTIN identifier. For example:
+
+``` 
+get_product(product_id) // GTIN 
+set_product(product_id, product) // GTIN, gs1Product 
+```
+
+### Product Addressing in the Merkle-Radix State System
+
+In order to uniquely locate GS1 products in the Merkle-Radix state system, an
+address must be constructed which identifies the storage location of the Product
+representation.
+
+All Grid addresses are prefixed by the 6-hex-character namespace prefix
+“621dee”,  Products are further prefixed under the Grid namespace with reserved
+enumerations of “02” (“00” and “01” being reserved for other purposes)
+indicating “Products” and an additional “01” indicating “GS1 Products”.
+
+Therefore, all addresses starting with:
+
+``` 
+“621dee” + “02” + “01” 
+```
+
+are Grid GS1 Products identified by a GTIN and are expected to contain a Product
+representation which conforms with the GS1 product schema.
+
+GTIN formats consist of 14-digit “numeric strings” which include some amount of
+internal “0” padding depending on the specific GTIN format (GTIN-8, GTIN-12,
+GTIN-13, or GTIN-14).  After the 10-hex-characters that are consumed by the grid
+namespace prefix, the product, and GS1 prefixes, there are 60 hex characters
```

Comment:
Is this (60 hex characters)  enough to accommodate for LGTIN (GTIN plus batch/lot and SGTIN (Serialized GTIN?) or is this out of scope?

Reply:
From the brief googling I did on LGTIN & SGTIN it seems that neither will exceed 60 characters in length. So yes, the length of a grid address is able to accommodate both.